standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -5484,6 +5509,29 @@ private void fireReadTablePreEvent(String catName, String dbName, String tblName
     }
   }
 
+  /**
+   * Fire a pre-event for read database operation, if there are any
+   * pre-event listeners registered
+   */
+  private void fireReadDatabasePreEvent(final String name)
+          throws MetaException, NoSuchObjectException {
+    if(preListeners.size() > 0) {
+      // do this only if there is a pre event listener registered (avoid unnecessary
+      // metastore api call)
```

Comment:
I think this comment is kinda redundant after `if(preListeners.size() > 0)`

Reply:
ack

Reply:
ack